pkg/controller/scheduler.go

```diff
@@ -4,26 +4,55 @@ import (
 	"fmt"
 	"time"
 
-	flaggerv1 "github.com/stefanprodan/flagger/pkg/apis/flagger/v1alpha2"
+	flaggerv1 "github.com/stefanprodan/flagger/pkg/apis/flagger/v1alpha3"
 	"k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// scheduleCanaries synchronises the canary map with the jobs map,
+// for new canaries new jobs are created and started
+// for the removed canaries the jobs are stopped and deleted
 func (c *Controller) scheduleCanaries() {
+	current := make(map[string]bool)
 	stats := make(map[string]int)
+
 	c.canaries.Range(func(key interface{}, value interface{}) bool {
-		r := value.(*flaggerv1.Canary)
-		if r.Spec.TargetRef.Kind == "Deployment" {
-			go c.advanceCanary(r.Name, r.Namespace)
+		canary := value.(*flaggerv1.Canary)
+		name := key.(string)
```

Comment:
nit: It'd be good to have a comment to show what these keys look like, it's hard to tell from the code. I'd assume it's something like `<namespace>/<name>`, but I'm not sure exactly.

Reply:
Yes it's `<namespace>.<name>` I'll add a comment thanks!